[FEATURE]: Enable a new versions endpoint which allows querying by creation time

fwilkens opened the pull request:
In #1266 , the request was made to extend the `just_updated` activities endpoint. I have a similar use-case in that I'd like to keep a system up to date with new version information as it becomes available periodically. In my case, the update client needs to be tolerant of outages/downtime with the ability to catch back up from around the time of outage. I checked out the webhooks api as was suggested in #1266 for this purpose, and while very cool, I think there's a legitimate use-case for an endpoint that allows getting versions that were created within a given timeframe.

An endpoint like this allows the client to skip dealing with tokens/auth, running a service to consume inbound webhooks, and dealing with webhook failure/retry scenarios.

Happy to elaborate or clarify if needed, thanks for considering this! API docs updated in https://github.com/rubygems/guides/pull/217.

fwilkens commented:
Hey there @sonalkr132, just checking in to see if there's anything I can do here to move this PR forward. Thanks!

fwilkens commented:
Hey @sonalkr132 or @dwradcliffe, I wanted to check in again to see if you have any feedback on this PR. Anything needed on my end before it can be reviewed? Any questions? Other blockers? Thanks for your time and have a great week!

sonalkr132 commented on app/controllers/api/v1/timeframe_versions_controller.rb:
```diff
@@ -0,0 +1,45 @@
+    return
+  end
+
+  def version_timeframe_query
```
Can we please move this to `Version` model and add unit tests for it? Also, perhaps a better name could be `created_between`.

fwilkens replied:
Sure thing, will do 👍

sonalkr132 commented on app/controllers/api/v1/timeframe_versions_controller.rb:
```diff
@@ -0,0 +1,45 @@
+  private
+
+  def from_time
+    @parse_from_time ||= Time.iso8601(params.require(:from))
```
why do we need `||=` here?

fwilkens replied:
It's just some memoization so that the parse only happens once, even though `from_time` is called twice.

sonalkr132 commented:
I don't this it would be wise to allow querying all version released since beginning of time. We will get DDOSed. 
It would be super helpful if you could benchmark this endpoint with our latest [data dump](https://rubygems.org/pages/data) for different time rangers (24hrs, 48hrs, 1 week etc).

fwilkens commented:
Updated @sonalkr132.

* Moved query into model w/ unit tests.
* Added a constraint for a 7 day span for any given query. Arguably anyone who needs larger timeframes can just use the database dumps, or compose separate api requests.

I ran some sampling locally with a recent database dump, browser page load for a 7 day timespan averages ~125ms on my machine. No measurable difference for smaller timeframes. I'll update the docs I wrote with this new constraint once this is approved. Thanks and have a good weekend!

sonalkr132 commented:
Thanks @fwilkens.
